### DISCUSSION TITLE
How do I get `&AppHandle` instance in `on_window_event` so that I can change system tray?

### DISCUSSION THREAD

**Discussion by erayerdin** [tauri-apps/tauri#4682] [Q&A] [answered]
Builder::[on_system_tray_event](https://docs.rs/tauri/1.0.4/tauri/struct.Builder.html#method.on_system_tray_event) method accepts a closure with signature containing the instance of `&AppHandle`.

AppHandle::[tray_handle](https://docs.rs/tauri/1.0.4/tauri/struct.AppHandle.html#method.tray_handle) method returns `SystemTrayHandle` wherein you can [set_menu](https://docs.rs/tauri/1.0.4/tauri/struct.SystemTrayHandle.html#method.set_menu). You can change system tray menu like this.

[Example code here](https://github.com/erayerdin/taurapp/blob/cf6a521ad4f5cc83da8fadd3d50b0aaf4c543023/src-tauri/src/tray.rs#L85)

However, the problem is Builder::[on_window_event](https://docs.rs/tauri/1.0.4/tauri/struct.Builder.html#method.on_window_event)'s closure signature. It does not have `&AppHandle` as instance. So I cannot do `app.tray_handle().set_menu(whatever)`.

How do I change system tray in `on_window_event` closure?

Thanks in advance.

***

# Environment

```plain
Environment
  › OS: Linux 20.04 X64
  › Node.js: 16.14.0
  › npm: 8.13.2
  › pnpm: Not installed!
  › yarn: 1.22.18
  › rustup: 1.24.3
  › rustc: 1.61.0
  › cargo: 1.61.0
  › Rust toolchain: stable-x86_64-unknown-linux-gnu 

Packages
  › @tauri-apps/cli [NPM]: 1.0.3 (outdated, latest: 1.0.4)
  › @tauri-apps/api [NPM]: 1.0.2
  › tauri [RUST]: 1.0.3,
  › tauri-build [RUST]: 1.0.3,
  › tao [RUST]: 0.12.1,
  › wry [RUST]: 0.19.0,

App
  › build-type: bundle
  › CSP: unset
  › distDir: ../dist
  › devPath: ../dist

App directory structure
  ├─ .git
  ├─ .github
  ├─ taurapp.AppDir
  ├─ dist
  ├─ node_modules
  ├─ .vscode
  └─ src-tauri
```

**Comment by FabianLars** [comment, marked as answer]
You can get an AppHandle via `event.window().app_handle()` if the `Manager` trait is imported (`use tauri::Manager;`
